### DISCUSSION TITLE
Laser engrave too slow

### DISCUSSION THREAD

**Discussion by nasp2000** [luc-github/ESP3D#952] [Questions]
I have been using esp3d version 3.0 on a laser with an 8-bit board. It's cool to have the laser working without a PC just through the ESP32. 

But there is one thing that ruins everything.... Engraving images with big gcodes of 4 or 5 MB the laser is very slow.

How can I resolve this? I tried diferent MicroSD but can solve this

**Comment by luc-github** [comment]
Unfortunatly current streaming method in esp3d V3 is quite basic and definitly not for fast jobs, what it would need it is a  buffered streaming (several commands sent) but current implementation is just a ping/pong streaming 1 commands / ack.

You can try to use a buffered streaming from a PC using raw TCP protocol, it may compensate network latency and also fit your needs.

In future I hope to implement buffered streaming but currently we are working on better ping pong streaming that also fit 3d printer commands
